src/AppInstallerCommonCore/Downloader.cpp

```diff
@@ -109,6 +109,12 @@ namespace AppInstaller::Utility
 
         dest.flush();
 
+        // Check download size matches if content length is provided in response header
+        if (contentLength > 0)
+        {
+            THROW_HR_IF(E_UNEXPECTED, bytesDownloaded != contentLength);

```

Comment:
>E_UNEXPECTED [](start = 24, length = 12)

let's create a new hresult reflecting the actual error of size mismatch
DOWNLOAD_SIZE_MISMATCH